protostar/testing/cairo_cheatcodes/call_cairo_cheatcode.py

```diff
@@ -0,0 +1,48 @@
+import asyncio
+from typing import Callable, Optional
+
+from protostar.starknet import CheatcodeException
+from protostar.starknet.cheaters.contracts import ContractsCheaterException
+from protostar.testing.cairo_cheatcodes.cairo_cheatcode import CairoCheatcode
+from protostar.starknet.data_transformer import CairoOrPythonData, CairoData
+from protostar.starknet import RawAddress, Address
+
+
+class CallCairoCheatcode(CairoCheatcode):
+    @property
+    def name(self) -> str:
+        return "call"
+
+    def build(
+        self,
+    ) -> Callable[[RawAddress, str, Optional[CairoOrPythonData]], CairoOrPythonData]:
+        return self.call
+
+    def call(
+        self,
+        contract_address: RawAddress,
+        function_name: str,
+        calldata: Optional[CairoOrPythonData] = None,
+    ) -> CairoData:
+        return asyncio.run(
+            self._call(
+                contract_address=Address.from_user_input(contract_address),
+                function_name=function_name,
+                calldata=calldata,
+            )
+        )
+
+    async def _call(
+        self,
+        contract_address: Address,
+        function_name: str,
+        calldata: Optional[CairoOrPythonData] = None,
+    ) -> CairoData:
+        try:
+            return await self.cheaters.contracts.call(
+                contract_address=contract_address,
+                function_name=function_name,
+                calldata=calldata,
+            )
+        except ContractsCheaterException as exc:
+            raise CheatcodeException(self, exc.message) from exc
```

Comment:
This method is probably redundant and we can do `asyncio.run(self.cheaters.contracts.call(...))` right away

Reply:
it's catching and re-throwing the caught exception though, right?

Reply:
Right, yeah we can leave it, it's not a big deal especially since we're going to refactor the error system anyway.